File: CWM.UI/cwm_desktop_mobile/lib/models/employee.dart
import 'package:json_annotation/json_annotation.dart';

import 'city.dart';
import 'country.dart';

part 'employee.g.dart';

@JsonSerializable()
class Employee {
  int id;
  String firstName;
  String lastName;
  DateTime birthDate;
  City? city;
  Country? citizenship;
  String email;
  String mobile;
  String adress;

  Employee(this.id, this.firstName, this.lastName, this.birthDate, this.city,
      this.citizenship, this.email, this.mobile, this.adress);

  factory Employee.fromJson(Map<String, dynamic> json) =>
      _$EmployeeFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeeToJson(this);
}

File: CWM.Services/Models/Employee.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CWM.Database.Models
{
    [Table("Employees")]
    public class Employee
    {
        [Key]
        public int Id { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Mobile { get; set; } = string.Empty;
        public DateTime BirthDate { get; set; }
        public string Email { get; set; } = string.Empty;
        public string Adress { get; set; } = string.Empty;

        [ForeignKey(nameof(City))]
        public int? CityId { get; set; }
        public virtual City? City { get; set; }

        [ForeignKey(nameof(Citizenship))]
        public int? CitizenshipId { get; set; }
        public virtual Country? Citizenship { get; set; }
        public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    }
}
